### DISCUSSION TITLE
How to build it?

### DISCUSSION THREAD

**Discussion by abidmalikwaterloo** [Snaipe/Criterion#505] [Q&A]
How can I build it on a Linux machine? I am using "` sudo meson install -C build`" and its not working

**Comment by Snaipe** [comment]
You'll have to provide more information than that 🙂

What does not work, what's your distro, what error message are you getting, etc etc.

That being said I think the rendering of the docs is broken, some commands are just not appearing. See the raw doc for building from source: https://raw.githubusercontent.com/Snaipe/Criterion/bleeding/doc/setup.rst

I assume you might not have run the meson setup & meson compile commands?

**Reply by crawfish-etouffee** [reply]
I am having the same issue as above. When I run the command `$ meson setup build`, I get the following error. 

```
The Meson build system
Version: 0.45.1
Source dir: /home/vagrant/Criterion
Build dir: /home/vagrant/Criterion/build
Build type: native build

meson.build:71:0: ERROR: lexer
        {'fn': 'clock_gettime'},
^
```

I am running Ubuntu 18.04.6 LTS

**Comment by Snaipe** [comment]
Also, converting this to a discussion, not a bug per-se.

**Comment by wesleyel** [comment]
```rst
Clone this repository:

.. code-block:: bash

    $ git clone https://github.com/Snaipe/Criterion.git

Then, run the following commands to build Criterion:

.. code-block:: bash
    $ meson setup build
    $ meson compile -C build
```

The following commands mentioned is missing in readthedocs, but existed in raw doc.

Maybe need a fix? @Snaipe 

https://[raw.githubusercontent.com/Snaipe/Criterion/bleeding/doc/setup.rst](https://raw.githubusercontent.com/Snaipe/Criterion/bleeding/doc/setup.rst)

https://criterion.readthedocs.io/en/master/setup.html#building-from-source
